kernel/src/process/posix_thread/exit.rs

```diff
@@ -143,8 +143,13 @@ fn wake_robust_list(thread_local: &ThreadLocal, tid: Tid) {
 
     trace!("exit: wake up the rubust list: {:?}", list_head);
     for futex_addr in list_head.futexes() {
-        let _ = wake_robust_futex(futex_addr, tid)
-            .inspect_err(|err| debug!("exit: cannot wake up the robust futex: {:?}", err));
+        if let Err(err) = wake_robust_futex(futex_addr, tid) {
+            debug!(
+                "exit: cannot wake robust futex at {:?}: {:?}",
+                futex_addr, err
+            );
+            return;
```

Comment:
I realized that this should be a `break` instead of a `return`? We can then ensure that the robust list is none and it makes sense since at least some futexes may already be woken up...

So either (1) change this to a `break`, or (2) change line 139 to `robust_list.take()` and remove the final `*robust_list = None`. It seems that the second option can result in simpler code.

This shouldn't have any practical impact because no one should access the robust list after the thread exit. I'm thinking that (1) having one more invariant for the dead process is slightly better and (2) if we clear the futex list after a full wakeup, then we should also clear it after a partial one.